### ISSUE TITLE
oneOf annotation has no effect on properties

### ISSUE THREAD

**Issue by maaland** [swagger-api/swagger-core#2900]
I have a `DTO` class containing a property that can be of different data types. After looking at [the spec](https://swagger.io/docs/specification/data-models/data-types/) I learned that using `oneOf` should help me achieve what I want. However, using the annotation does not seem to have the desired effect on the generated `.json`. 

I have a class that looks like this:

```
@Schema(name = "SomeDTO")
public class SomeDTO {
  @Schema(description = "Description of someLongField", )
  private Long someLongField;
  @Schema(
      description = "Description of value",
      oneOf = { String.class, Number.class }
  )
  private JsonPrimitive value;
}
```

I expected the generated `.json` file to contain something like this:

```
"components" : {
  "schemas": {
    "SomeDTO": {
        "type": "object",
        "properties": {
          "someLongField": {
            "type": "integer",
            "description": "Description of someLongField.",
            "format": "int64"
          },
          "value": {
            "description": "Description of value.",
            "oneOf": [
              {
                "type": "string"
              },
              {
                "type": "number"
              }
            ]
          }
        }
      }
  }
}
```

While what I get is this:

```
"components" : {
  "schemas": {
    "SomeDTO": {
        "type": "object",
        "properties": {
          "someLongField": {
            "type": "integer",
            "description": "Description of someLongField.",
            "format": "int64"
          },
          "value": {
             "$ref": "#/components/schemas/JsonPrimitive"
          }
        }
      }
  }
}
```

With `JsonPrimitive` looking something like this:

```
"JsonPrimitive" : {
        "type" : "object",
        "properties" : {
          ... bunch of default properties
        },
        "description" : "The data value. Can be String or numeric depending on the metric",
        "oneOf" : [ { }, { } ]
      },
```

Am I using the `oneOf` annotation incorrectly, or is this a bug/missing feature?

Any help is greatly appreciated.

**Comment by frantuma** [comment]
Thanks for reporting this! on one side, there was a bug causing "primitive" types not to be resolved, this is fixed in #2902, please try with latest snapshot from sonatype.

At the moment, all properties resolved as objects are resolved as reference to an object defined in `components` section, therefore it is not possible (except using filters or extensions as described in https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Extensions, e.g. by providing a custom resover) to have the property defined inline as in your expected output. such support is planned but no ETA at the moment.

In your scenario you are also using a POJO (Gson `JsonPrimitive`) for which (if I get this right) you don't want some properties to show up. Check  [this example](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-core/src/test/java/io/swagger/v3/core/resolving/resources/TestObjectTicket2900.java) (with related [test](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-core/src/test/java/io/swagger/v3/core/resolving/ComposedSchemaTest.java)) for 2 ways to implement this without applying filters or extension, basically you can define your own POJO as `implementation` and annotate that, or ignore `JsonPrimitive` properties at Jackson level via mixins.

**Comment by maaland** [comment]
Thanks for the quick response @frantuma !
Unfortunately, I'm not really at liberty to rely on snapshots in my application. I also need my dependencies to be in maven central. Any ETA on the release of `2.0.3` in central?

If I understand you correctly, the solution you propose in your last paragraph relies on #2902 , right?
